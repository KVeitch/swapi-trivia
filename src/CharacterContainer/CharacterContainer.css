.character__container-main {
  display:flex;
  height: 70vh;
  width:85%;
  justify-content:center;
  align-self:center;
}

/* Crawl Stylings */

.character__container-crawl {
  background:rgba(0, 0, 0, 0.7);
  width:100%;
  height:70vh;
  overflow: hidden;
}

.fade {
  position: relative;
  width: 100%;
  min-height: 30vh;
  top: -25px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1;
}

.crawl__text {
  display: flex;
  justify-content: center;
  position: relative;
  height: 60vh;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 300%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  perspective: 400px;
  text-align: justify;
}

.crawl {
  position: relative;
  top: 99999px;
  transform-origin: 50% 100%;
  animation: crawl 60s linear;
}

/* End of Crawl styles  */

.character__container {
  height: 70vh;
  overflow: scroll;
}

.div__image {
  align-self:center;
  height: 262px;
  width: 480px
}

.loading__text {
  align-items: center;
  color: rgba(67, 52, 45, 0.7);
  font-size: 1.7em;
  font-family: 'Barlow Condensed', sans-serif;
  text-shadow: 1px 2px 3px rgba(23, 6, 1, 0.94);
}

.character__container {
  align-self: center;
  border: 1px solid rgba(67, 52, 45, 0.7);
  display: flex;
  flex-wrap: wrap;
  width: 85%;
}

@keyframes crawl {
	0% {
    top: -100px;
    transform: rotateX(20deg)  translateZ(0);
  }
	100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}