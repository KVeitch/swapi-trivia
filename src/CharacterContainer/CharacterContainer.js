import React from 'react';
import CharacterCard from '../CharacterCard/CharacterCard';

const CharacterContainer = props => {
    const characterList = props.characters.map((character, i) => {
        console.log('character', character)
        return <CharacterCard 
                    key={i} 
                    character={character} />
    })
    return (
        <section className='character__container'>
            {characterList}
        </section>
    )
}

export default CharacterContainer;