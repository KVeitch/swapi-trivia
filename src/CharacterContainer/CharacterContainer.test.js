import React from 'react';
import { shallow } from 'enzyme';
import CharacterContainer from './CharacterContainer';

describe('Character Container', () => {
    let wrapper
    let mockChangeSelectedMovie = jest.fn();
    let mockSetFavorite = jest.fn();

    it('should match the snapshot with all the data passed in correctly', () => {
        wrapper =
            shallow(<CharacterContainer
                characters={[]}
                changeSelectedMovie={mockChangeSelectedMovie}
                isReady={false}
                setFavorite={mockSetFavorite}
                favoriteList={[]}/>)
        expect(wrapper).toMatchSnapshot();
    });
});
