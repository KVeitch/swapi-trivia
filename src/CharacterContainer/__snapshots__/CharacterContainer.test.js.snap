// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Container should match the snapshot with all the data passed in correctly 1`] = `
<section
  className="character__container"
/>
`;
